Add suggestion for user stories

hendrikb opened the pull request:
Hey Folks,

have been starting to sketch our way from "Project Description" to "User Stories" to "Tasks" to "Architecture/Tech" to "Stuff that can be coded". Feel free to add User Stories until we have everything covered that our application(s) should do.

Best,
Hendrik

hendrikb committed:
Add suggestion for user stories

omarsotillo commented on UserStories.md:
```diff
@@ -0,0 +1,8 @@
+# User Stories
+
+## Controling
```
```suggestion
## Controlling
```

omarsotillo commented on UserStories.md:
```diff
@@ -0,0 +1,8 @@
+
+### Backend
+
+* A Backend User wants to register, login and log out from a Backend
```
Let's try to avoid having technical wording I would say in the use cases. For that, I suggest using the [MoSCoW method](https://en.wikipedia.org/wiki/MoSCoW_method). 

Example:
- An unprotected user/authority must be able to log in, register and logout.
- A user must be of type unprotected (opened a request/issue for needed topic) or authority
- An unprotected user must be allowed to raise an issue for any kind of topic.
- A topic must contain the categories of shopping help, the conflicting situation at home.
- If the category is conflicting situation at home allow specifying communication method
- A topic could contain the category official news for cities
- A topic should be ...
- Some topics must notify authorities.
- Some topics must show information.
- Authority users should be able to define a notification interval for an unprotected user.
- Authority users should be able to notify users specific decisions (messages) taken by them and the next steps.
- Authority users must be able to assign a status for the different requests
- A request status must be processing, finished, to_be acted.
- Authority users must define critical level of request 
- A request critical level should be critical, medium, low,
- A request has a topic decided by the unprotected user.

hendrikb replied:
yes, thanks for refining, completely makes sense. feel free to integrate your suggestions into the file already and then we move forward to establish a shared understanding of what should be achieved. :)

hendrikb replied:
yo @omarsotillo - check this out https://github.com/TheRealWolkenbruch/concept/pull/1/commits/3eef7f1f151ad25a53d4fe9fed68e7b45279609f - I've been adding "my stuff" and I think I kinda have described what I wanted to achieve with this tool, very roughly. You had a bunch of great additions to the service "catalog", feel free to add them to the User Stories, I didn't dare to turn your thoughts into text. :)

hendrikb replied:
So, to wrap our meeting up: I encourage @doggygit @Garllon @Thudilie to ALSO add their wishes to the user stories. @Garllon said that he wants to create a Google Doc to work on the Use Cases quicker, but the results should be here in the end. :)

hendrikb committed:
Fix Headlines in UserStories.md

Co-Authored-By: Omar Sotillo <kurodasogo@gmail.com>

hendrikb committed:
Sketch "Backend Login" & "DeadMansSwitch Service"

Tried to follow @omarsotillo recommendation and used MoSCoW notation and
left some placeholders for further "service" defintions & open
questions.

Garllon commented on UserStories.md:
```diff
@@ -0,0 +1,42 @@
+General Idea:
+
+The whole idea that I'm proposing here is that Unprotected Users don't need a
+login/credential to be able to use the service. The usage of the service must
```
Any idea how we should do this. That the user can click the dead mans switch without logged in?

hendrikb replied:
what about a URL with a hash? We could put that into a reoccurring calendar event or a reminder? Keeping things simple.

Thudilie replied:
should the backend generate the hash or should the hash be some kind of combined information of name, email, etc?
If the hash is some kind of combination the hash could be long or? Is request body here an option?
(This happens on button click then i think.)

hendrikb replied:
i think we can just "define" it and then work with it. I have a slight favor for that hash being "complicated" so it cannot be guessed + it does not reveal anything about the protected person. what do you think?

Thudilie replied:
I'm a bit curious how the definition of the transformed hash would look like. 

correct me if my thoughts for the normal flow are wrong: 

So the unprotected user needs to make the initial setup:
The data for the setup is something like name, email
- after doing the initial setup in the register gui and submitting the data, the hash comes as an response from the backend and is saved in local storage, cookie or so. 
- Then the unprotected user can kind of use the 'i'm not dead'-report button.
with every button click the user sets url-params with the hash that was created at the beginning

doggygit replied:
That is the flow I have also in my mind.

But I am also thinking: If we are doing this with a hash, how does Login/Logout work? Because my understanding is then that the unprotected user would be just gone.

hendrikb replied:
So, again, to wrap things up, we discussed that only Authority Users should have a real username/pw login. Unprotected Users should receive "well known links" to reset their timers and edit their data.

These links are sent via Email (at least in this MVP; options for mobile Apps later). They should change eventually or should at least have some sort of "best before date".

Garllon replied:
https://docs.google.com/document/d/1fIZqcX8eQHx6z39_F-8e90HbP4IdakSEK87_5CTBumA/edit?usp=sharing

I wrote my first ideas her

hendrikb replied:
I took the major parts of your gdocs suggestions @Garllon - thank you. Therefor I am resolving this thread. :)

Thudilie commented on UserStories.md:
```diff
@@ -0,0 +1,42 @@
+### Backend
+
+* An Authority User must be able to register, login and log out from a Backend
+* An Unprotected User should be able to register, login and log out from a Backend
```
The frontend gui should provide this right?

hendrikb replied:
I think so, I have only weak opinions regarding frontend/backend software architecture, though. Most modern apps would be API-only backend-wise and basically everything that involves "HTML and things" sits as a front end application on top of that, right? i like i discuss options here. :)

Thudilie replied:
sounds legit :D leave it that way!

Thudilie replied:
@hendrikb

Thudilie commented on UserStories.md:
```diff
@@ -0,0 +1,42 @@
+
+### Frontend
+
+TBD: Also - how does a frontend to an Unprotected User look like?
```
how about this?
https://drive.google.com/file/d/1lzqZhNhhAbWeSeQ-KC3gSdMzNcCHNuP7/view?usp=sharing
very basic though 
Feel free to comment and i will adjust :)

hendrikb replied:
greeeeat first work, this goes into the right directions @Thudilie!! i bet we can work with that!

hendrikb replied:
I put the description + the sketch image into the UserStories in d90aa67bd9fbdb664d105e1f98408e9a71b44a18 - so I am resolving this. :)

hendrikb commented:
Everyone, feel free to just change lines in the concept document and we'll discuss the changes there. It's easier to do it like that compared to discuss everything in the comments and then "transfer" the outcomes back into the docs.

hendrikb committed:
Set next meeting: 18:00 Tuesday

hendrikb committed:
Set next Meeting for Thursday 16:30

hendrikb committed:
Set project name to AreYouThere

fixes #3

hendrikb committed:
Set next meetings on Friday & Saturday

hendrikb committed:
Adjust UserStories after 2020-04-23 call

hendrikb committed:
Add Frontend sketch

hendrikb committed:
Fix Frontend Sketch image

hendrikb committed:
Merge branch 'master' into user_stories

hendrikb commented:
You think this is good? Then hit the button!

omarsotillo commented:
Looking good to me :) You tell me to hit the button and that is what I do 👍 @hendrikb  🚀
